### DISCUSSION TITLE
Qual o papel do profissional de frontend na sua empresa?

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#428] [Discussão]
Uma questão cultural e que engloba aspectos distintos de cada empresa é o papel de um perfil na organização. E o que me chama a atenção é o papel do frontend hoje no Brasil em algumas empresas. Então queria saber de vocês qual é esse papel do front end nas suas empresas. Podem também adicionar outras coisas que vocês acham importante que ele deveria estar fazendo. Coisas que vocês discordam também e tal.

**Comment by fernandosouza** [comment]
Opa, Edmo Lima. Uma ótima pergunta. É de meu total interesse em conhecer como os times de FrontEnd são formados em diferentes empresas.
Posso citar 2 papeis diferentes de FrontEnds existentes no escritório onde trabalho. Acredito que essa divisão pode ser similar em empresas de medio e grande porte, porém um tanto diferente em empresas de pequeno porte. Segue:

1 - UI FrontEnd Engineer. Cria componentes com base em freamworks e os utiliza para compor as interfaces.
2 - Infrastructure FrontEnd Engineer. É responsável por criar e manter frameworks e dar suporte a demais necessidades que os UI FrontEnd Engineers necessitam para trabalhar mais efetivamente. Por exemplo, estrutura de testes unitários e CI. 

Para ambos os papeis podemos encontrar perfis diferentes de profissionais. Aqueles que são mais especializados em CSS, aqueles que possuiem conhecimento tanto em CSS quanto JS bem como em BackEnd, e aqueles que não possuiem conhecimento algum em CSS mas que são experts em programação, apenas escrevem JavaScript e também conseguem trabalhar como BackEnds.

O ecosistema FrontEnd é extremamente vasto, e acredito que por isso seja comum essa divercidade de competências técnicas. Eu acredito que quão mais versátio um profissional é, melhor. Por ter uma visão macro do problema. Porém, ter essa diversidade técnica em um mesmo time é muito salutar. A experiência de um completa a falta de expriência do outro.

Espero que tenha respondido tua pergunta.

**Comment by ghost** [comment]
Legal @fernandosouza !

Aqui na empresa também temos diferentes papéis e é bem similar ao que tu citou. Isso é legal por que temos diferentes perspectiva, ui e de engenharia de software mesmo. Respondeu sim e gostaria também de conhecer mais como os profissionais frontend se organizam e qual os seu papel nos times. Hoje aqui onde trabalho o nosso papel tá bem voltado a padronização de interfaces e de design de código. Trabalhamos em um modelo adaptado do "spotify" de guilds, então temos esses diferentes perfis (UI e de Engineer).

Espero que outros profissionais possam contribuir =)

**Comment by felipefialho** [comment]
Aqui na Filtro, como agência digital:

### Principais Tarefas

- Transformar layouts estáticos em interfaces ricas
- Definição de Arquitetura CSS 
- Fornecer bagagem em usabilidade
- Cuidar das acessibilidade
- Cuidar do SEO
- Garantir funcionamento nos browsers definidos

### Tarefas Secundárias

- Implementar e testar novas tecnologias client-side.
- Ajudar no back-end para implementar novas features no sistema
- Fazer QA e aplicar os ajustes
- Ajudar com o que mais for preciso

---

Nossa área de atuação é realmente ampla...
